### DISCUSSION TITLE
Extensions not working as expected.

### DISCUSSION THREAD

**Discussion by Charmainpayne** [coder/code-server#6848] [General]
Hey guys,

I’m a newbie and i have recently setup Code server on my unraid media server/docker.  I have searched  the group how also with no suggestions.

I’m in the middle of setting up sqltools and i wanted to create a new connection. When I click on the button a window apprears but nothing shows up like it should. See the photo.

I have also tried using the codeium chat but when the window appears it’s blank also.  I’ve attached a photo to explain what I’m seeing.

Could i get some help?

![IMG_0157](https://github.com/coder/code-server/assets/99071080/604ca916-da47-469a-a687-d5b7b72fbe00)

**Comment by code-asher** [comment]
Are you using https?  Usually when web views like these are not working it is because of http.  Web views rely on service workers, and service workers can only be used in secure contexts.  There are settings to disable that though, I believe.  In Chrome for example there is the `unsafely-treat-insecure-origin-as-secure` setting.

If you are using https, please post the logs from the browser console, there might be something there to help debug.

**Reply by Charmainpayne** [reply]
Thanks for responding. I haven't set up my PFSense router yet or Nginx so that makes sense. I therefore haven't setup https just yet. I'll get that sorted.
